### ISSUE TITLE
Yunohost Package for element-call

### ISSUE THREAD

**Issue by Gredin67** [YunoHost-Apps/element_ynh#42]
Element provides now native webRTC support for p2p group calls. No need for external jitsi instance anymore.
Does it make sense to create an external YNH package? Or can you add it to this package as an option?

https://github.com/vector-im/element-call

**Comment by Josue-T** [comment]
Hello,

I would prefer to integrate this in the actual element package. I didn't know this, thanks for the report.

**Comment by rosbeef** [comment]
I prefer integrated too enabled by default but with an option to disable it in config panel and/or in install panel

**Comment by Josue-T** [comment]
Or maybe it would prefer to integrate in the synapse package so we have the server part in synapse and element stay the client part.

**Comment by rosbeef** [comment]
is someone tried to package it ?

as it not depends neither "only" on element neither on synapse it should be a separated package. it should work in the future with other serveur and have an autonomous interface so no need to element.

Having an external package and an option in elements to indicates call service could be more flexible.

**Comment by Gredin67** [comment]
@YunoHost-Apps/matrix-bridges  what do you think of this ? Did some of you already try element-call on its main synapse ?

> Element Call requires a homeserver with registration enabled without any 3pid or token requirements, if you want it to be used by unregistered users. Furthermore, it is not recommended to use it with an existing homeserver where user accounts have joined normal rooms, as it may not be able to handle those yet and it may behave unreliably.
> 
> Therefore, to use a self-hosted homeserver, this is recommended to be a new server where any user account created has not joined any normal rooms anywhere in the Matrix federated network. The homeserver used can be setup to disable federation, so as to prevent spam registrations (if you keep registrations open) and to ensure Element Call continues to work in case any user decides to log in to their Element Call account using the standard Element app and joins normal rooms that Element Call cannot handle.

**Comment by Thatoo** [comment]
What does "normal rooms anywhere in the Matrix federated network" mean?

**Comment by Gredin67** [comment]
Obviously, there is no backward compatibility of rooms (version) supporting element-call.
Or maybe there are now two rooms type without compatibility at all : the historical ones for chatting, and new ones for videoconf.

What I understand is that they are talking about the use-case of a jitsi-like conferencing system, where any non-authenticated user can join from web with a shared url link or so.

But our use-case is more the whatsapp/signal-like video calls, where all users are authenticated. Two things to be clarified though:
- Federation question: Can users authenticated on another homeserver join the call on an element-call that is linked to our homeserver ?
- Bridge question: do we want people on a whatsapp/signal side to be able to join the call --> this would obviously require to have a second public homeserver separate from the one of the bridges.

**Comment by MayeulC** [comment]
> What does "normal rooms anywhere in the Matrix federated network" mean?

I think they mean Element-call does not really support text rooms yet.
I'm quite busy, so I haven't looked into Matrix video calling yet, but there are multiple `m.room.type` that were proposed for MSCs recently (spaces are one type of room).

I imagine those "call" rooms are a specific type of room, and this version of element does not support regular rooms.

> Federation question: Can users authenticated on another homeserver join the call on an element-call that is linked to our homeserver ?

I think so. But if you want to allow unregistered users to join the call, you need open registrations to make them an account.

> Bridge question: do we want people on a whatsapp/signal side to be able to join the call --> this would obviously require to have a second public homeserver separate from the one of the bridges.

Do bridges even support calls? :exploding_head:

**Comment by Thatoo** [comment]
Hoping I'm wrong, I understand that we'll need to install a dedicated server (synapse, dendrite or some day conduit...) with element-call.
So we'll need two matrix servers, the main one (the one we installed so far) and one dedicated to call. Clients (Element Web, Element Android etc...) will have to embed Element Call widget from the url of the second matrix server (how will they know the url of second one, I guess thanks to .well-known.).

**Comment by dinosmm** [comment]
Hi everyone, I (re)wrote the part of the README you are quoting there. I am just a regular user who likes to self-host things; I wrote that to clarify the situation for others, since I found the previous text not helpful enough for me (I got very confused as to what Element Call could do when I tried to self-host it).

All it means is that, right now, Element Call needs a Matrix homeserver to work. If you only want authenticated users to make *and join* calls, that's fine, it will do that with any homeserver.
If you want to be able to accept guests to join calls if they are not authenticated / Matrix users, it will only work if your homeserver allows open registration (without requirement for email or phone number or token).

> Federation question: Can users authenticated on another homeserver join the call on an element-call that is linked to our homeserver ?

I didn't try to ask someone not on my homeserver to join my call, but I think it would work fine.

> What does "normal rooms anywhere in the Matrix federated network" mean?

I meant Matrix rooms not created via Element Call. 
Originally, the devs said Element Call may not handle it well if the HS already has rooms and users in those rooms (so, if it is an existing HS). In my tests, I found that Element Call took a while to start working properly, but once it did, it worked fine even on a HS with existing rooms, so this advice *may* be outdated.

Please note that the Element apps don't yet have Element Call integrated, as far as I know this is planned for the coming weeks but is not yet completed.

**Comment by Thatoo** [comment]
Thank you @dinosmm it is much clearer for me.
I have one question though. Is it on the roadmap to be able to have a matrix homeserver that don't accept registration (only ldap authentification) , let authenticated users launch element calls and be able to invite guests thanks to a url (previously given)?
Can we install element-call now and expect this feature to come later on?

**Comment by dinosmm** [comment]
This is my own understanding from my experience - I am not a Matrix dev or even an expert, so might be wrong:

Synapse currently does support some sort of guest access, but as far as I know Element does *not* have any guest functionality. Element Call is the same - it does not support adding unregistered guests to calls, even if Synapse might support it.

I am not sure if there is any Matrix client out there that supports unregistered guests, and I have no idea if the Matrix team plan to introduce this functionality into Element or Element Call any time soon unfortunately.

If you want this functionality right now, the only (?) way is to run Element Call on a separate Synapse server, which has open registration. For security (and peace of mind) it might be a good idea for that Synapse server to be unfederated (so even if a spammer registers, they cannot use it to spam anyone outside that server).

**Comment by MayeulC** [comment]
> Please note that the Element apps don't yet have Element Call integrated, as far as I know this is planned for the coming weeks but is not yet completed.

Hmm, I thought that's what the video room beta was for?

<details><summary>element desktop screenshot</summary>

![image](https://user-images.githubusercontent.com/3952726/220930764-a49f5fb2-3c48-4419-866f-25d880c3a647.png)
</details>

Well... Personally, I don't care too much about allowing unregistered guests to join video calls for now.

**Comment by dinosmm** [comment]
> Hmm, I thought that's what the video room beta was for?

As per the link below, Video Rooms are still based on Jitsi (so basically they are videocall-only rooms with no other difference to having a normal Matrix room and placing a group call within it).
When Element Call gets integrated, they will flip Video Rooms to use Element Call instead.

https://element.io/blog/drop-in-drop-out-chats-with-video-rooms-and-a-new-search-experience/

**Comment by Thatoo** [comment]
If you want, you can test Element call inside Element if you add 
```
{
    "showLabsSettings": true
}
```

in (linux) /home/USER/.config/Element/config.json

restart Element and then go into "Expérimental" again, then you'll find below

![image](https://user-images.githubusercontent.com/464782/220935956-3c0d8201-f692-4f86-95e3-2f6cb8c0a5b5.png)

**Comment by ericgaspar** [comment]
closing -> https://github.com/YunoHost-Apps/element-call_ynh
